aimmo-game/tests/functional/test_effect_expiry.py

```diff
@@ -0,0 +1,169 @@
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 10)
+        self.assertEqual(self.avatar.attack_strength, 11)
+
+        # Move twice to the second pickup.
+        for i in range(2):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertTrue(isinstance(list(self.avatar.effects)[1], pickup_created_two.EFFECT))
+        self.assertEqual(len(self.avatar.effects), 2)
+        self.assertEqual(self.avatar.attack_strength, 26)
+
+        # Eight turns later, we expect the first effect to expire.
+        for i in range(8):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 2)
+        self.assertEqual(self.avatar.attack_strength, 16)
+
+        # Two turns later, the second pickup expires too.
+        for i in range(2):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(len(self.avatar.effects), 0)
+        self.assertEqual(self.avatar.attack_strength, 1)
+
+    def test_multiple_invulnerability_boost_pickup_expiry(self):
+        """
+        Ensure that each invulnerability boost effect expires at the appropriate time, even if the user stacks up on
+        them. One pickup will be at (1,0) and another at (3,0).
+        """
+        cell_two = self.game.game_state.world_map.get_cell(Location(3, 0))
+        pickup_created_one = InvulnerabilityPickup(self.cell)
+        pickup_created_two = InvulnerabilityPickup(cell_two)
+
+        self.cell.pickup = pickup_created_one
+        cell_two.pickup = pickup_created_two
+
+        self.assertEqual(self.avatar.resistance, 0)
+
+        # Avatar moves EAST to (1,0) where pickup one is located.
+        self.game.turn_manager._run_single_turn()
+
+        self.assertTrue(isinstance(list(self.avatar.effects)[0], pickup_created_one.EFFECT))
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 10)
+        self.assertEqual(self.avatar.resistance, 1000)
+
+        # Move twice to the second pickup.
+        for i in range(2):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertTrue(isinstance(list(self.avatar.effects)[1], pickup_created_two.EFFECT))
+        self.assertEqual(len(self.avatar.effects), 2)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 8)
+        self.assertEqual(list(self.avatar.effects)[1]._time_remaining, 10)
+        self.assertEqual(self.avatar.resistance, 2000)
+
+        # Eight turns later, we expect the first effect to expire.
+        for i in range(8):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 2)
+        self.assertEqual(self.avatar.resistance, 1000)
```

Comment:
same

Reply:
That's the resistance effect that the InvulnerabilityPickupEffect causes the avatar to have.